src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1,139 +1,2148 @@
+
+enum QuestTheGilneasLiberationFront
+{
+    NPC_FORSAKEN_TROOPER1 = 44791,
+    NPC_FORSAKEN_TROOPER2 = 44792,
+    NPC_WORGEN_RENEGATE = 44793,
+
+    SPELL_HEARTSTRIKE = 84182,
+    SPELL_DARKENED = 73307,
+    SPELL_FLURRY_OF_CLAWS_CHANNEL = 80365,
+    SPELL_FLURRY_OF_CLAWS = 80367,
+
+    EVENT_CHECK_SHOWFIGHT = 101,
+    EVENT_SPELL_QUEUE = 102,
+    EVENT_SPELL_DAMAGE = 103,
+
+    MOVE_TO_HOMEPOSITION,
+
+    TALK_WORGEN_DEFEATED = 0
+};
+
+// Worgen Renegade - 44793
+struct npc_silverpine_worgen_renegade : public ScriptedAI
+{
+    npc_silverpine_worgen_renegade(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        _events.Reset();
+
+        me->RemoveAura(SPELL_DARKENED);
+
+        _events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 1s);
+    }
+
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    {
+        if (!caster)
+            return;
+
+        if (spell->Id == SPELL_HEARTSTRIKE)
+            me->AddAura(SPELL_DARKENED, me);
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type == POINT_MOTION_TYPE && pointId == MOVE_TO_HOMEPOSITION)
+            _events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 25ms);
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.ScheduleEvent(EVENT_SPELL_QUEUE, 3s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CHECK_SHOWFIGHT:
+                {
+                    if (me->IsInCombat() || me->isDead())
+                        return;
+
+                    if (me->GetDistance2d(me->GetHomePosition().GetPositionX(), me->GetHomePosition().GetPositionY()) > 10.0f)
+                    {
+                        me->GetMotionMaster()->MovePoint(MOVE_TO_HOMEPOSITION, me->GetHomePosition());
+                        return;
+                    }
+
+                    if (Creature* forsaken = me->FindNearestCreature(NPC_FORSAKEN_TROOPER1, 5.0f))
+                    {
+                        me->Attack(forsaken, true);
+                        return;
+                    }
+                    else if (Creature* forsaken = me->FindNearestCreature(NPC_FORSAKEN_TROOPER2, 5.0f))
+                    {
+                        me->Attack(forsaken, true);
+                        return;
+                    }
+                    else
+                        _events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 2s + 500ms);
+                    break;
+                }
+
+                case EVENT_SPELL_QUEUE:
+                {
+                    DoCastVictim(SPELL_FLURRY_OF_CLAWS_CHANNEL);
+
+                    _events.ScheduleEvent(EVENT_SPELL_QUEUE, 16s, 18s);
+                    break;
+                }
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+        else
```

Comment:
`else` not needed

if code returns, DoMeleeAttackIfReady is never reached

Reply:
Fixed.